### PR TITLE
Split updown to risefall and higherlower

### PR DESCRIPTION
**Pull request by qingweibinary**
This changes affects quite a number of things, (code is short though)

I've tried to make sure it is safe, but still we need a careful review, @borisyankov @nuruddeensalihu help me take a look and hope I did not miss anything out

### CODE REVIEW

**Comment by qingweibinary** [comment]
This PR is about several things as follow

1. Fix all race condition I've encounter, basically the root cause is that asset picker can change the trade params, thus trade params lose some of the control, and behave incorrectly as the code is develop as a self-contained entity but grow into something more complex.

2. Remove redundant subscription call, the root cause is the same with number 1.

Note: we need to redesign this part (After stable release ;) )

File: src/_reducers/TradingOptionsReducer.js [hunk 1]
```diff
@@ -3,10 +3,38 @@ import { UPDATE_TRADING_OPTIONS } from '../_constants/ActionTypes';
...
+                }
+            }
+        } else return opt;
+    });
+
 export default (state = initialState, action) => {
     switch (action.type) {
         case UPDATE_TRADING_OPTIONS: {
-            return state.set(action.symbol, action.options);
+            const enrichedOptions = splitRiseFallAndHighLow(action.options);
```
Comment (borisyankov) [inline src/_reducers/TradingOptionsReducer.js]: Oh man, this naming ;)

File: src/fulltrade/FullTradeParams.js [hunk 2]
```diff
@@ -28,10 +34,13 @@ import { createDefaultType, createDefaultDuration, createDefaultBarriers } from
...
  * 5. barriers are non available for contract below 2 minutes
- * 6. forward starting does not have barriers
+ * 6. For underlying that's under our control, we allow relative and absolute barrier regardless of intraday or not
+ * 7. Relative barrier is always allowed
+ * 8. Forward starting do not have barriers
+ * 9. All underlying have risefall trade
  *
  *
- * TODO: reform the whole dependency mess,
+ * TODO: re-architect the whole dependency mess,
```
Comment (borisyankov) [inline src/fulltrade/FullTradeParams.js]: 10. All assumptions converted to unit tests :)

Comment (qingweibinary) [inline src/fulltrade/FullTradeParams.js]: most of this assumption is assumption on data, meaning we are short of metadata, we need to confirm with Quants, and think what to do when assumption breaks

**Pull request merged**